internal/host/hostcommands/install_cmd.go

```diff
@@ -290,10 +293,35 @@ func constructHostInstallerArgs(cluster *common.Cluster, host *models.Host, inve
 		}
 	}
 
-	installerArgs, hasIPConfigOverride := appends390xArgs(inventory, installerArgs, log)
+	installerArgs, hasIPConfigOverride = appends390xArgs(inventory, installerArgs, log)
 
 	hasUserConfiguredIP := hasUserConfiguredIP(installerArgs)
 
+	// append kargs depending on installation drive type
+	installationDisk := hostutil.GetDiskByInstallationPath(inventory.Disks, hostutil.GetHostInstallationPath(host))
+	if installationDisk != nil {
+		installerArgs = appendMultipathArgs(installerArgs, installationDisk)
+		installerArgs, err = appendISCSIArgs(installerArgs, installationDisk, inventory, hasUserConfiguredIP)
+		if err != nil {
+			return "", err
+		}
+
+		// When using ISCSI along OCI, we expect the user (via a
+		// script) to configure the network statically on the nodes as
+		// DHCP is not available in this case.
+		//
+		// Even if the user configured the network during discovery, we
+		// cannot propagate the configuration with --copy-network
+		// because it won't work along iSCSI:
+		// https://github.com/coreos/coreos-installer/issues/1389
+		hasIPConfigOverride = hasIPConfigOverride ||
+			(installationDisk.DriveType == models.DriveTypeISCSI && common.IsOciExternalIntegrationEnabled(cluster.Platform))
+	} else {
+		log.Warnf("No installation disk found for host ID %s", host.ID)
+	}
+
+	hasStaticNetwork := (infraEnv != nil && infraEnv.StaticNetworkConfig != "") || cluster.StaticNetworkConfigured
+
 	// set DHCP args only if no IP config override was specified (only for LPAR and zVM nodes on s390x)
 	if !hasStaticNetwork && !hasIPConfigOverride && !hasUserConfiguredIP {
```

Comment:
This `hasStaticNetwork` shown twice should be moved to a common 

also line 326 look smilar maybe we can combine this to a single check

Reply:
refactored, let me know!